app/helpers/badge_helper.rb

```diff
@@ -23,17 +23,21 @@ def parole_review_date?(offender)
     offender.indeterminate_sentence? && offender.parole_review_date.present?
   end
 
-  def early_allocation_notes?(offender, early_allocations)
-    if early_allocations.any?
-      !offender.within_early_allocation_window? || early_allocations.last.community_decision == false
+  def early_allocation_notes?(sentence, early_allocations)
+    allocations = current_sentence_allocations(sentence, early_allocations)
+    if allocations.any?
+      !allocations.last.created_within_referral_window? || allocations.last.community_decision == false
     end
   end
 
-  def early_allocation_active?(early_allocations)
-    early_allocations.any? && early_allocations.last.awaiting_community_decision?
+  def early_allocation_active?(sentence, early_allocations)
+    allocations = current_sentence_allocations(sentence, early_allocations)
+    allocations.any? && allocations.last.awaiting_community_decision?
   end
 
-  def early_allocation_approved?(early_allocations)
-    early_allocations.any? && early_allocations.last.created_within_referral_window? && early_allocations.last.community_decision_eligible_or_automatically_eligible?
+private
+
+  def current_sentence_allocations(sentence, early_allocations)
+    early_allocations.select { |early_allocation| early_allocation.assessment_date.after?(sentence.sentence_start_date) }
```

Comment:
This filtering is handled by the model, so we should be using the model method and deleting this